tests/pipelines/animatediff/test_animatediff_video2video.py

```diff
@@ -0,0 +1,269 @@
+import unittest
+
+import numpy as np
+import torch
+from PIL import Image
+from transformers import CLIPTextConfig, CLIPTextModel, CLIPTokenizer
+
+import diffusers
+from diffusers import (
+    AnimateDiffVideoToVideoPipeline,
+    AutoencoderKL,
+    DDIMScheduler,
+    MotionAdapter,
+    UNet2DConditionModel,
+    UNetMotionModel,
+)
+from diffusers.utils import is_xformers_available, logging
+from diffusers.utils.testing_utils import torch_device
+
+from ..pipeline_params import TEXT_TO_IMAGE_PARAMS, VIDEO_TO_VIDEO_BATCH_PARAMS
+from ..test_pipelines_common import PipelineTesterMixin
+
+
+def to_np(tensor):
+    if isinstance(tensor, torch.Tensor):
+        tensor = tensor.detach().cpu().numpy()
+
+    return tensor
+
+
+class AnimateDiffVideoToVideoPipelineFastTests(PipelineTesterMixin, unittest.TestCase):
+    pipeline_class = AnimateDiffVideoToVideoPipeline
+    params = TEXT_TO_IMAGE_PARAMS
+    batch_params = VIDEO_TO_VIDEO_BATCH_PARAMS
+    required_optional_params = frozenset(
+        [
+            "num_inference_steps",
+            "generator",
+            "latents",
+            "return_dict",
+            "callback",
+            "callback_steps",
```

Comment:
```suggestion
            "callback",
            "callback_steps",
            "callback_on_step_end",
            "callback_on_step_end_tensor_inputs",
```

Reply:
Not able to test the changes locally unfortunately. Seem to be getting errors from huggingface_hub:

<details>
<summary>Error log</summary>

```bash
__________________________________________________________________ ERROR collecting tests/pipelines/animatediff/test_animatediff_video2video.py ___________________________________________________________________
src/diffusers/utils/import_utils.py:693: in _get_module
    return importlib.import_module("." + module_name, self.__name__)
/usr/lib/python3.10/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1050: in _gcd_import
    ???
<frozen importlib._bootstrap>:1027: in _find_and_load
    ???
<frozen importlib._bootstrap>:1006: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:688: in _load_unlocked
    ???
<frozen importlib._bootstrap_external>:883: in exec_module
    ???
<frozen importlib._bootstrap>:241: in _call_with_frames_removed
    ???
src/diffusers/pipelines/animatediff/pipeline_animatediff_video2video.py:36: in <module>
    from ..pipeline_utils import DiffusionPipeline
src/diffusers/pipelines/pipeline_utils.py:38: in <module>
    from huggingface_hub.utils import OfflineModeIsEnabled, validate_hf_hub_args
E   ImportError: cannot import name 'OfflineModeIsEnabled' from 'huggingface_hub.utils' (/home/arrow/Desktop/projects/github/diffusers/.venv/lib/python3.10/site-packages/huggingface_hub/utils/__init__.py)

The above exception was the direct cause of the following exception:
.venv/lib/python3.10/site-packages/_pytest/runner.py:341: in from_call
    result: Optional[TResult] = func()
.venv/lib/python3.10/site-packages/_pytest/runner.py:372: in <lambda>
    call = CallInfo.from_call(lambda: list(collector.collect()), "collect")
.venv/lib/python3.10/site-packages/_pytest/python.py:531: in collect
    self._inject_setup_module_fixture()
.venv/lib/python3.10/site-packages/_pytest/python.py:545: in _inject_setup_module_fixture
    self.obj, ("setUpModule", "setup_module")
.venv/lib/python3.10/site-packages/_pytest/python.py:310: in obj
    self._obj = obj = self._getobj()
.venv/lib/python3.10/site-packages/_pytest/python.py:528: in _getobj
    return self._importtestmodule()
.venv/lib/python3.10/site-packages/_pytest/python.py:617: in _importtestmodule
    mod = import_path(self.path, mode=importmode, root=self.config.rootpath)
.venv/lib/python3.10/site-packages/_pytest/pathlib.py:565: in import_path
    importlib.import_module(module_name)
/usr/lib/python3.10/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1050: in _gcd_import
    ???
<frozen importlib._bootstrap>:1027: in _find_and_load
    ???
<frozen importlib._bootstrap>:1006: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:688: in _load_unlocked
    ???
.venv/lib/python3.10/site-packages/_pytest/assertion/rewrite.py:178: in exec_module
    exec(co, module.__dict__)
tests/pipelines/animatediff/test_animatediff_video2video.py:9: in <module>
    from diffusers import (
<frozen importlib._bootstrap>:1075: in _handle_fromlist
    ???
src/diffusers/utils/import_utils.py:684: in __getattr__
    value = getattr(module, name)
src/diffusers/utils/import_utils.py:684: in __getattr__
    value = getattr(module, name)
src/diffusers/utils/import_utils.py:683: in __getattr__
    module = self._get_module(self._class_to_module[name])
src/diffusers/utils/import_utils.py:695: in _get_module
    raise RuntimeError(
E   RuntimeError: Failed to import diffusers.pipelines.animatediff.pipeline_animatediff_video2video because of the following error (look up to see its traceback):
E   cannot import name 'OfflineModeIsEnabled' from 'huggingface_hub.utils' (/home/arrow/Desktop/projects/github/diffusers/.venv/lib/python3.10/site-packages/huggingface_hub/utils/__init__.py)

```
</details>

Reply:
Nvm, I was using the wrong virtual environment carried from a different branch :sweat_smile: